### ISSUE TITLE
conda run is not preserving exit code

### ISSUE THREAD

**Issue by impredicative** [conda/conda#9599]
## Current Behavior
`conda run` used to return the correct exit status of the target process, but it doesn't anymore. This was apparently working correctly in conda 4.8.0 but it seems broken since version 4.8.1. `conda run` now erroneously always return the exit status of 0. This is a serious problem as jobs fail silently now.

### Steps to Reproduce
```
$ conda -V
conda 4.8.1
$ conda run -n myenv /usr/bin/true; echo $?
0
$ conda run -n myenv /usr/bin/false; echo $?
ERROR conda.cli.main_run:execute(30): Subprocess for 'conda run ['/usr/bin/false']' command failed.  (See above for error)
0
```

## Expected Behavior
`conda run` should preserve the exit status of the target process, whether the target is `python` or otherwise. It worked correctly with the previous version:
```
$ conda -V
conda 4.8.0
$ conda run -n base /usr/bin/true; echo $?
0
$ conda run -n base /usr/bin/false; echo $?
ERROR conda.cli.main_run:execute(39): Subprocess for 'conda run ['/usr/bin/false']' command failed.  Stderr was:

1
```

## Environment Information
This issue really is independent of the environment as it happens both in various dockerfiles and on MacOS too.

---

@msarahan Please fix this bug considering I see a commit from 25 days that may have introduced it! Adding unit tests would also be wise.

**Comment by mingwandroid** [comment]
Our team have many things to work on. PRs that yet to fix this would be welcome if you are interested. Conda run is still not ready for prime time usage unfortunately.

**Comment by impredicative** [comment]
@mingwandroid This used to work correctly just a month ago. Failing silently is one of the worst kind of failures since it leads to errors in production that no one finds out about in time. If this won't be fixed, I will have to stop using `conda run`.

**Comment by mingwandroid** [comment]
We are not churning out new conda features.

**Comment by impredicative** [comment]
> PRs that try to fix this would be welcome

@gilescope Just in case there is interest. Thanks.

**Comment by mlline00** [comment]
It appears commit https://github.com/conda/conda/commit/c7d5ada5d0a6038e81f9ab10380e5e69c7d65d1c introduced the error.

Returning a namedtuple (Response) instead of an int breaks this logic https://github.com/conda/conda/blob/e9a50561880696e57ba80472dce332e08e58ee0b/conda/cli/main.py#L84-L86

I can create a pull request to fix. Thoughts on reverting the return value to an int vs expanding the logic at the call site in main.py?  I can't find rationale for changing the return value.

**Comment by impredicative** [comment]
The scarier point as I see it is that `conda run` being experimental is an excuse for the conda team to break it willy-nilly and to not bother to fix it, to eschew responsibility. Even if this issue is fixed, I cannot trust using `conda run` while it stays experimental. IIRC, `conda init` is also experimental, and I guess I can't use that either in my production code.

I now use a simple [PATH based approach](https://stackoverflow.com/a/57595180/) instead in my Dockerfile.

**Comment by impredicative** [comment]
FWIW, here is a high-level unit test to test the functionality after a fix is implemented:

```python
import subprocess
import unittest


class TestExitCodes(unittest.TestCase):
    # Note: conda=4.8.1 (but not conda=4.8.0) always incorrectly returns exit code 0.
    def test_exit_codes(self):
        for expected_exit_code in range(3):
            with self.subTest(exit_code=expected_exit_code):
                actual_exit_code, _output = subprocess.getstatusoutput(f"conda run exit {expected_exit_code}")
                self.assertEqual(expected_exit_code, actual_exit_code)


if __name__ == "__main__":
    unittest.main()
```

If using this test with a failing conda, e.g. v4.8.1., consider adding the decorator `    @unittest.expectedFailure` to the test.

**Comment by esc** [comment]
I'd like to see a fix for this also.

**Comment by esc** [comment]
I believe this is a duplicate of #8385 perhaps?

**Comment by impredicative** [comment]
Instead of finding duplicates going back a year, how about providing information about a fix (along with unit tests to guard against future screwups), so we can then close out these issues? Or did Anaconda lay off everyone already? Just what are the managers prioritizing if not critical bug fixes like this one?

**Comment by jjhelmus** [comment]
@impredicative I have removed you latest comment from this issue.  It is not constructive and the tone is not welcome here.  If you continue to make comments along these lines your account will be banned.

**Comment by impredicative** [comment]
Closing duplicate/pointless issue.

**Comment by esc** [comment]
Could someone perhaps re-open this issue. This isn't a duplicate of #8385 after all. But a regression of that issue. There also already seems to be  a PR to fix both at #9665 .

**Comment by impredicative** [comment]
Finally this is fixed and released in v4.8.4. Thank you @chenghlee !

`conda run` however continues to remain experimental and therefore unfit for production use.
